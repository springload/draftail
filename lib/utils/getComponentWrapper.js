import React from "react";

/**
 * Wraps a component to provide it with additional props based on context.
 */
const getComponentWrapper = (Wrapped, wrapperProps) => {
  const Wrapper = (props) => <Wrapped {...props} {...wrapperProps} />;

  return Wrapper;
};

export default getComponentWrapper;
