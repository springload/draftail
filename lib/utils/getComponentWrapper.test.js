import React from "react";
import { shallow } from "enzyme";

import getComponentWrapper from "./getComponentWrapper";

describe("getComponentWrapper", () => {
  it("works", () => {
    const Wrapped = () => <div />;
    const Wrapper = getComponentWrapper(Wrapped, {
      test: true,
    });

    expect(shallow(<Wrapper />)).toMatchSnapshot();
  });
});
