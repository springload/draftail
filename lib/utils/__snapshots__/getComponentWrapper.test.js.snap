// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentWrapper works 1`] = `
<Wrapped
  test={true}
/>
`;
